tests/helpers/test_entity_platform.py

```diff
@@ -1710,14 +1710,16 @@ async def generate_response(
     }
 
 
-async def test_invalid_entity_id(hass: HomeAssistant) -> None:
+async def test_invalid_entity_id(
+    hass: HomeAssistant, caplog: pytest.LogCaptureFixture
+) -> None:
     """Test specifying an invalid entity id."""
     platform = MockEntityPlatform(hass)
     entity = MockEntity(entity_id="invalid_entity_id")
-    with pytest.raises(HomeAssistantError):
-        await platform.async_add_entities([entity])
+    await platform.async_add_entities([entity])
     assert entity.hass is None
     assert entity.platform is None
+    assert "Invalid entity ID: invalid_entity_id" in caplog.text
```

Comment:
This test relied on races adding entities. In the sensor tests we expect it won't fail, but here we expected it to fail